PhysicsTools/NanoAOD/plugins/GlobalVariablesTableProducer.cc

```diff
@@ -76,13 +76,16 @@ class GlobalVariablesTableProducer : public edm::stream::EDProducer<> {
   class Variable {
   public:
     Variable(const std::string& aname, const edm::ParameterSet& cfg)
-        : name_(aname), doc_(cfg.getParameter<std::string>("doc")) {}
+        : name_(aname),
+          doc_(cfg.getParameter<std::string>("doc")),
+          precision_(cfg.existsAs<int>("precision") ? cfg.getParameter<int>("precision") : -1) {}
```

Comment:
if the parameter is not optional (preferred, and more logical in the flow of the code);
```suggestion
          precision_(cfg.getParameter<int>("precision") ) {}
```